### PR TITLE
fix(dropbear_luks_unlock): Add config paths for debian 12

### LINKED ISSUE #159: Dropbear-initramfs config paths changed in Deb 12
The dropbear_path changed from `/etc/dropbear-initramfs` to `/etc/dropbear/initramfs`
please update the paths inside the roles accordingly
On a sidenote: The task `Configure /etc/crypttab` tries to write to a file `/etc/dropbear-initramfs/config` which doesn't exist (anymore?) and the path should be changed to the actual location of crypttab.
Thankies :3

### PR DESCRIPTION
**Pull request by jadyndev**
Changes as suggested by @jdreichmann.
resolve #159

### CODE REVIEW

File: roles/dropbear_luks_unlock/defaults/main.yml [hunk 1]
```diff
@@ -3,8 +3,19 @@ dropbear_luks_required_packages:
...
+dropbear_crypttab_path: "/etc/crypttab"
+dropbear_initramfs_base_paths:
+  Debian:
+    10: "/etc/dropbear-initramfs"
+    11: "/etc/dropbear-initramfs"
+    12: "/etc/dropbear/initramfs"
+  default: "/etc/dropbear-initramfs"
+
+dropbear_initramfs_base_path: "{{ dropbear_initramfs_base_paths[ansible_distribution]
+                                                               [ansible_distribution_major_version | int]
```
Comment (jdreichmann) [inline roles/dropbear_luks_unlock/defaults/main.yml:15]: I have done the opposite, having String indices, as f.ex. 11.5 or a release name in other distros might not be an integer, while an integer can be represented in a string. Also, have not yet found a good way to have a "ansible_distribution" level default...

Comment (jdreichmann) [reply roles/dropbear_luks_unlock/defaults/main.yml:15]: Also, it is more common to have strings as indices in yaml dicts. both work though

Comment (jadyndev) [reply roles/dropbear_luks_unlock/defaults/main.yml:15]: hmm yeah release names could resolve this, but that'll fall apart on rolling release OS and OS without proper names. For `Debian` this *should* work tho.

Comment (jdreichmann) [reply roles/dropbear_luks_unlock/defaults/main.yml:15]: True, but i would change the 10, 11, 12 to "10", "11", "12" and cast to string, would then also work for release code names without changing much of the logic. Arch's `distribution_major_version` is probably just not set..?

Comment (jadyndev) [reply roles/dropbear_luks_unlock/defaults/main.yml:15]: Sounds reasonable. Will change asap

**Review by jcgruenhage — APPROVED** [review]
Looks good to me. Under the assumption that this has been tested on Debian 11/12 and also *some* other distro to check for the default fallback, I'd want to see this merged.

**Comment by jadyndev** [comment]
After testing on the latest Debian 11 and Debian 12 I can confirm that the paths are configured correctly
```cli
jadyn@testing-deb11:/$ tree /etc/dropbear-initramfs/
/etc/dropbear-initramfs/
|-- config
|-- dropbear_ecdsa_host_key
|-- dropbear_ed25519_host_key
`-- dropbear_rsa_host_key

0  directories, 4 files
```

```cli
jadyn@testing-deb12:~$ sudo tree /etc/dropbear/initramfs/
/etc/dropbear/initramfs/
|-- authorized_keys
|-- dropbear.conf
|-- dropbear_ecdsa_host_key
|-- dropbear_ed25519_host_key
`-- dropbear_rsa_host_key

1 directory, 5 files
```

**Review by jcgruenhage — APPROVED** [review]

**Comment by jadyndev** [comment]
Further testing showed that ubuntu-20.04, ubuntu-22.04 and ubuntu-23.04 also use `/etc/dropbear/initramfs/` which is why we should make this the default and only handle the debian <=11 as an edge case

> After testing on the latest Debian 11 and Debian 12 I can confirm that the paths are configured correctly
> 
> ```
> jadyn@testing-deb11:/$ tree /etc/dropbear-initramfs/
> /etc/dropbear-initramfs/
> |-- authorized_keys
> |-- config
> |-- dropbear_ecdsa_host_key
> |-- dropbear_ed25519_host_key
> `-- dropbear_rsa_host_key
> 
> 0  directories, 4 files
> ```
> 
> ```
> jadyn@testing-deb12:~$ sudo tree /etc/dropbear/initramfs/
> /etc/dropbear/initramfs/
> |-- authorized_keys
> |-- dropbear.conf
> |-- dropbear_ecdsa_host_key
> |-- dropbear_ed25519_host_key
> `-- dropbear_rsa_host_key
> 
> 1 directory, 5 files
> ```

**Review by jcgruenhage — APPROVED** [review]

**Pull request merged**